reports/src/main/java/com/ivy/reports/ReportFlowViewModel.kt

```diff
@@ -0,0 +1,413 @@
+    private val groupTrnsFlow: GroupTrnsFlow,
+    private val exportCSVLogic: ExportCSVLogic,
+) : FlowViewModel<ReportState, ReportState, ReportsEvent>() {
+
+    private val filter: MutableStateFlow<FilterState> = MutableStateFlow(emptyFilterState())
+    private val filterVisible = MutableStateFlow(false)
+
+    private val loading = MutableStateFlow(false)
+    private val trnsList = MutableStateFlow<List<Transaction>>(emptyList())
+
+    private val _categoryNone = dummyCategory(
+        name = "None",
+        color = Gray.toArgb(),
+        icon = dummyIconSized(R.drawable.ic_custom_category_s)
+    )
+
+    init {
+        initialiseData()
+    }
+
+    override fun initialState(): ReportState = emptyReportScreenState("")
+
+    override fun stateFlow(): Flow<ReportState> = combineMultiple(
+        baseCurrencyFlow(), headerStateflow(), grpTrnsListFlow(), filter, filterVisible, loading
+    ) { baseCurr, headerState, grpTrnsList, filterState, visible, loading ->
+
+        ReportState(
+            baseCurrency = baseCurr,
+
+            headerState = headerState.toImmutableItem(),
+            trnsList = grpTrnsList.toImmutableItem(),
+
+            filterVisible = visible,
+            filterState = filterState,
+
+            loading = loading
+        )
+    }
+
+    private fun headerStateflow() = trnsFlow()
+        .flatMapLatest { list ->
+            baseCurrencyFlow().flatMapMerge { currencyCode ->
+                calculateWithTransfersFlow(
+                    CalculateWithTransfersFlow.Input(
+                        trns = list,
+                        outputCurrency = currencyCode,
+                        accounts = getSelectedAccounts()
+                    )
+                )
+            }.map { transactionStats ->
+                HeaderState(
+                    balance = transactionStats.balance,
+                    income = transactionStats.income,
+                    expenses = transactionStats.expense,
+
+                    incomeTransactionsCount = transactionStats.incomesCount,
+                    expenseTransactionsCount = transactionStats.expensesCount,
+
+                    treatTransfersAsIncExp = false,
+                    showTransfersAsIncExpCheckbox = false,
+
+                    transactionsOld = list.toOld().toImmutableItem(),
+                    accountIdFilters = getSelectedAccounts().map { a -> a.id }
+                        .toImmutableItem()
+                )
+            }
+        }
+        .onStart { emit(emptyHeaderState()) }
+
+    private fun grpTrnsListFlow() = trnsFlow()
+        .flatMapLatest {
+            groupTrnsFlow(it)
+        }
+        .onStart { emptyTransactionList() }
+
+    private fun trnsFlow() = filter.flatMapLatest {
+        reportsFilterTrnsFlow(
+            ReportsFilterTrnsFlow.Input(
+                filterState = it,
+                noneCategory = _categoryNone
+            )
+        )
+    }.onEach { list ->
+        trnsList.value = list
```

Comment:
this is an ugly side-effect? why do we need it?

Reply:
We need this Exports,

Either have this or store it in state and fetch via state

Reply:
Have it in state but as transaction new and map trns new to trns old when the export btn is clicked